src/Microsoft.CodeQuality.Analyzers/VisualBasic/Maintainability/BasicRemoveUnusedLocals.vb

```diff
@@ -75,11 +75,13 @@ Namespace Microsoft.CodeQuality.VisualBasic.Analyzers.Maintainability
                         Sub(operationContext)
                             Dim localReferenceExpression As ILocalReferenceOperation = DirectCast(operationContext.Operation, ILocalReferenceOperation)
                             Dim syntax = localReferenceExpression.Syntax
-
-                            ' The writeonly references with non-invocation right-hand side should be ignored.
                             If syntax.Parent.IsKind(SyntaxKind.SimpleAssignmentStatement) Then
                                 Dim parent = DirectCast(syntax.Parent, AssignmentStatementSyntax)
-                                If parent.Left Is syntax AndAlso parent.Right.Kind() <> SyntaxKind.InvocationExpression Then
+                                Dim simpleAssignmentExpression As IAssignmentOperation = DirectCast(localReferenceExpression.Parent, IAssignmentOperation)
```

Comment:
Definitely, I know less than you know about VB syntax. :-) Could you please provide an example? I'll add it to unit tests. Thanks!

---
In reply to: [152688029](https://github.com/dotnet/roslyn-analyzers/pull/1336#discussion_r152688029) [](ancestors = 152688029)

Reply:
```
Class C
    Sub M()
        Dim a As Short = 0
        a = 1
    End Sub
End Class
```

Reply:
Actually, for the above case you should still be fine as the conversion is on the value (right). I think it might still be better to rewrite the code as below:
```
Dim parent = DirectCast(syntax.Parent, AssignmentStatementSyntax)
If parent.Left Is syntax Then
  Dim simpleAssignmentExpression As IAssignmentOperation = DirectCast(localReferenceExpression.Parent, IAssignmentOperation)
  If simpleAssignmentExpression.Value.ConstantValue.HasValue Then
    Return
  End If
End If
```

This will avoid cast exception for cases where local reference expression is on the right hand side of an assignment.

Reply:
I think your current code with throw case exception on below:

```
Class C
    Sub M()
        Dim a As Short = 0, b As Integer = 0
        a = b
    End Sub
End Class
```